src/steps/smart-campaign-add-lead.ts

```diff
@@ -0,0 +1,55 @@
+/*tslint:disable:no-else-after-return*/
+
+import { BaseStep, Field, StepInterface } from '../core/base-step';
+import { Step, FieldDefinition, StepDefinition } from '../proto/cog_pb';
+
+export class AddLeadToSmartCampaignStep extends BaseStep implements StepInterface {
+
+  protected stepName: string = 'Add Marketo Lead to Smart Campaign';
+  protected stepExpression: string = 'add the (?<email>.+) marketo lead to smart campaign (?<campaign>.+)';
+  protected stepType: StepDefinition.Type = StepDefinition.Type.ACTION;
+  protected expectedFields: Field[] = [{
+    field: 'email',
+    type: FieldDefinition.Type.EMAIL,
+    description: 'the email address of the Marketo Lead',
+  },
+  {
+    field: 'campaign',
+    type: FieldDefinition.Type.ANYSCALAR,
+    description: 'either the name of the smart campaign, or its numeric ID',
+  }];
+
+  async executeStep(step: Step) {
+    const stepData: any = step.getData().toJavaScript();
+    const email = stepData.email;
+    const campaign = stepData.campaign;
+    let campaigns = [{ id:campaign, isRequestable:true }];
+
+    try {
+      if (isNaN(campaign)) {
+        const allCampaigns = await this.client.getCampaigns();
+        campaigns = allCampaigns.result.filter(c => c.name.toLowerCase() == campaign.toLowerCase());
+        console.log(campaigns);
+      }
+
+      if (campaigns.length != 1) {
+        return this.error("Can't add %s to %s: found more than one campaign with that name.", [email, campaign[0]]);
```

Comment:
- [ ] Could be erroneous/confusing if there were 0 campaigns found with the name.  Proposed alternative message: `Can't add %s to %s: found %d matching campaigns`
- [ ] Also seems like the second string would not be replaced with the right value (`campaign[0]`)